sugoi-api-ldap-store-provider/src/main/java/fr/insee/sugoi/store/ldap/LdapReaderStore.java

```diff
@@ -318,14 +319,15 @@ private SearchResultEntry getEntryByDn(String dn) {
       LdapMapper<ResultType> mapper)
       throws LDAPSearchException {
     SearchRequest searchRequest = new SearchRequest(baseDn, scope, filter, "*", "+");
-    LdapUtils.setRequestControls(searchRequest, pageableResult);
-    SearchResult searchResult = ldapPoolConnection.search(searchRequest);
+    LdapUtils.setRequestControls(searchRequest, pageableResult, config);
+    SearchResult searchResult = ldapMonoConnection.search(searchRequest);
     PageResult<ResultType> pageResult = new PageResult<>();
-    List<ResultType> results =
+    pageResult.setResults(
         searchResult.getSearchEntries().stream()
             .map(e -> mapper.mapFromAttributes(e.getAttributes()))
-            .collect(Collectors.toList());
-    pageResult.setResults(results);
+            .collect(Collectors.toList()));
+    LdapUtils.setResponseControls(pageResult, searchResult);
```

Comment:
if we have all the results at first time totalElements will be set to 0, I don't know if it is a problem but maybe

Reply:
how can you know that all results are retrieved ? without vlv we cannot know that all elements were found